session.go

```diff
@@ -750,6 +750,51 @@ func (s *Session) Destroy(w http.ResponseWriter) error {
 	return nil
 }
 
+// DestroyUsingTokenPrefix deletes all token from storage that matches
+// the token prefix. NOTE: this function will also delete current
+// session if the token matches the tokenPrefix
+func (s *Session) DestroyUsingTokenPrefix(w http.ResponseWriter) error {
+	if s.loadErr != nil {
+		return s.loadErr
+	}
+
+	if s.tokenPrefix == "" {
+		return fmt.Errorf("tokenPrefix is not set. You can call session.SetTokenPrefix(..) to set token prefix")
+	}
+
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	pattern := s.tokenPrefix + PrefixTokenDelimiter
+	err := s.store.DeleteByPattern(pattern)
+	if err != nil {
+		return err
+	}
+
+	// Used to prevent the case were the token was deleted from storage,
+	// but is still inside the cookie.
+	if strings.Contains(s.token, pattern) {
```

Comment:
Why do we need this check? Is it possible that this token has the prefix set in the session object but not in its token?

Reply:
Technically it is possible to set a prefix that the token doesn't have. You would do something like this:

```
session.SetPrefix("test")
session.Renew() // Creates new token
session.SetPrefix("other")
session.DestroyByPrefix()
```
Now if we don't have the check we'd just delete the token from session, but it would still exist in storage.